pandas/core/frame.py

```diff
@@ -6244,12 +6244,20 @@ def explode(self, column: Union[str, Tuple]) -> "DataFrame":
         if not self.columns.is_unique:
             raise ValueError("columns must be unique")
 
-        result = self[column].explode()
-        return (
-            self.drop([column], axis=1)
-            .join(result)
-            .reindex(columns=self.columns, copy=False)
-        )
+        result = self.copy()
+        exploded_col = result.pop(column).reset_index(drop=True).explode()
+        result = result.reset_index().join(exploded_col)
+
+        if isinstance(self.index, ABCMultiIndex):
```

Comment:
this is not great here, what exactly are you trying to do?

Reply:
I'm trying to check whether the index is a multiindex or not - saw this in other parts of the module, e.g.
```
                if isinstance(self.index, ABCMultiIndex):
                    # array of tuples to numpy cols. copy copy copy
                    ix_vals = list(map(np.array, zip(*self.index.values)))
                else:
                    ix_vals = [self.index.values]
```

Reply:
of course, that’s not what i mean

it is completely o scrub what you are doing
pls show me an example of the index before and after for a MI and an Indez

the purpose of the .explode()  is to exactly handle this
this likely needs to be directly handled in the cython code instead

Reply:
we do not want to special case MI (other codes are generally just older )

Reply:
not sure I understand what you're proposing @jreback. The original implementation calls `explode` in the `Series` as it does in here, so it's calling the cython version, but when joining it joins by the index and if there are duplicate entries in the index it does an unwanted cross-product.

Actually, it just came to my mind, that we could make things much simpler if instead of resetting the index to join on it without duplicates, we create a temporary column with a range and we join with it. That would make things much simpler and less hacky, and there is no need to touch the index. Does that make sense to you?

Reply:
@jreback thanks for your review - I've removed the special case handling

@datapythonista where would that temporary column go though (and how would I name it without risking overwriting an existing column)? I can't add this temporary range column to `self` _before_ calling `explode`, and then I would be exploding a dataframe, and that's exactly the method I'm trying to fix, and I don't see how adding it after helps

Reply:
you're right, what I said doesn't make sense, I got confused, sorry